### PR TITLE
source expression resolves to Int in call activity

### PR DESCRIPTION
**Pull request by im-pratham**
#### Check List:
* Unit tests:  Yes 
* Documentation: NA

PR is to fix situation where Integer value of source expression is saved as String in Target Expression of call activity

### CODE REVIEW

**Comment by filiphr** [comment]
Thanks for the PR @im-pratham. Can you please add a test case for this change? It would be good to see where the problem is and why this fixes it

**Comment by im-pratham** [comment]
@filiphr added test case to cover added functionality let me know if more is required,

File: modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.flowable.common.engine.impl.javax.el.ELContext;
+import org.flowable.common.engine.impl.javax.el.ELResolver;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (dbmalkovsky) [inline modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java:12]: Per the project ide settings all static imports are listed first after the package name; thanks.

File: modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+class JsonNodeELResolverTest {
+    private final ELResolver resolver = new JsonNodeELResolver();
+    private final ELContext context = new FlowableElContext(resolver, null);
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    void integerValueShouldBeIntNode() {
+        ObjectNode node = mapper.createObjectNode();
+        resolver.setValue(context, node, "age", 11);
+        assertTrue(node.get("age").isInt());
```
Comment (dbmalkovsky) [inline modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java:26]: The project's standard testing framework is assertj; please see https://assertj.github.io/doc/

File: modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
+package org.flowable.common.engine.impl.el;
```
Comment (dbmalkovsky) [inline modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java:1]: Please add the standard license header; thanks.

File: modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (dbmalkovsky) [inline modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java:3]: The import order is defined in https://github.com/flowable/flowable-engine/tree/master/ide-settings:

- statics
- java
- javax
- org
- com
- liquibase
- everything else

File: modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java (hunk 1 above)
Comment (im-pratham) [reply modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java]: Done

File: modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java (hunk 2 above)
Comment (im-pratham) [reply modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java]: Done

File: modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java (hunk 3 above)
Comment (im-pratham) [reply modules/flowable-engine-common/src/test/java/org/flowable/common/engine/impl/el/JsonNodeELResolverTest.java:13]: done

**Review by filiphr — CHANGES_REQUESTED** [review]
Thanks for adding the tests @im-pratham. However, we do not want to test on such a low level. For us in Flowable it is important that the functionality works on the API level.

For you it seems to be a problem when you are writing in a nested Json Variable. I would suggest looking into [`JsonTest`](https://github.com/flowable/flowable-engine/blob/master/modules/flowable-engine/src/test/java/org/flowable/engine/test/json/JsonTest.java) and adding some of the missing test cases there.

Some tests that might inspire you are:

* `testSetNestedJsonNodeValue`
* `testSetDateInJsonNode`
* `testSetInstantInJsonNode`
* `testSetArrayNestedJsonNodeValue`
* `testSetNestedArrayNestedJsonNodeValue`

**Comment by im-pratham** [comment]
> Thanks for adding the tests @im-pratham. However, we do not want to test on such a low level. For us in Flowable it is important that the functionality works on the API level.
> 
> For you it seems to be a problem when you are writing in a nested Json Variable. I would suggest looking into [`JsonTest`](https://github.com/flowable/flowable-engine/blob/master/modules/flowable-engine/src/test/java/org/flowable/engine/test/json/JsonTest.java) and adding some of the missing test cases there.
> 
> Some tests that might inspire you are:
> 
>     * `testSetNestedJsonNodeValue`
> 
>     * `testSetDateInJsonNode`
> 
>     * `testSetInstantInJsonNode`
> 
>     * `testSetArrayNestedJsonNodeValue`
> 
>     * `testSetNestedArrayNestedJsonNodeValue`

That is what I was looking for. I will commit on test case fot it

**Comment by im-pratham** [comment]
@filiphr made changes. Have a look

**Comment by filiphr** [comment]
Thanks for doing the changes @im-pratham. I have manually integrated this in https://github.com/flowable/flowable-engine/commit/83f721c794fb9f0629f01df004d1ed8f4bbd9239

**Pull request closed without merging**